File: examples/java-app/src/main/java/example/javaapp/Reader.java
package example.javaapp;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public final class Reader {

	private Reader() {}

	public static List<String> readLines(String path) {
		try {
			String content = Files.readString(Path.of(path));
			return Arrays.stream(content.split("\n"))
					.filter(line -> !line.trim().isEmpty())
					.collect(Collectors.toList());
		} catch (IOException e) {
			return List.of("alpha", "beta", "gamma", "delta");
		}
	}
}

File: examples/rust-app/src/reader.rs
use std::fs;

pub fn readLines(path: &str) -> Vec<String> {
	let content = fs::read_to_string(path);
	if let Ok(text) = content {
		return text
			.lines()
			.filter(|line| !line.trim().is_empty())
			.map(|line| line.to_string())
			.collect();
	}

	vec![
		"alpha".to_string(),
		"beta".to_string(),
		"gamma".to_string(),
		"delta".to_string(),
	]
}
